Bug: Undefined symbol `_roundeven`

#10413 introduced `Number#round_even` which binds to the LLVM intrinsic `roundeven` which was only introduced in LLVM 11. For older LLVM versions it falls back to `rint`.
It seems that the intrinsic is not (yet) available on darwin, for some reason.

----

I can't compile the compiler anymore :-(

```
Undefined symbols for architecture x86_64:
  "_roundeven", referenced from:
      _*Float64#round_even:Float64 in F-loat64.o
```

This is on Mac OSX, LLVM 11.1.0

_Originally posted by @asterite in https://github.com/crystal-lang/crystal/issues/10474#issuecomment-792021817_

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -510,6 +510,9 @@ jobs:
 
 workflows:
   version: 2
+  test_all_platforms:
+    jobs:
+      - test_darwin
   tagged_release:
     jobs:
       - test_linux:

--- a/src/float.cr
+++ b/src/float.cr
@@ -156,7 +156,7 @@ struct Float32
   def round_even : self
     # LLVM 11 introduced llvm.roundeven.* intrinsics which may replace rint in
     # the future.
-    {% if compare_versions(Crystal::LLVM_VERSION, "11.0.0") >= 0 %}
+    {% if compare_versions(Crystal::LLVM_VERSION, "11.0.0") >= 0 && !flag?(:darwin) %}
       LibM.roundeven_f32(self)
     {% else %}
       LibM.rint_f32(self)
@@ -257,7 +257,7 @@ struct Float64
   def round_even : self
     # LLVM 11 introduced llvm.roundeven.* intrinsics which may replace rint in
     # the future.
-    {% if compare_versions(Crystal::LLVM_VERSION, "11.0.0") >= 0 %}
+    {% if compare_versions(Crystal::LLVM_VERSION, "11.0.0") >= 0 && !flag?(:darwin) %}
       LibM.roundeven_f64(self)
     {% else %}
       LibM.rint_f64(self)

--- a/src/float.cr
+++ b/src/float.cr
@@ -154,13 +154,9 @@ struct Float32
   # Rounds towards the nearest integer. If both neighboring integers are equidistant,
   # rounds towards the even neighbor (Banker's rounding).
   def round_even : self
-    # LLVM 11 introduced llvm.roundeven.* intrinsics which may replace rint in
-    # the future.
-    {% if compare_versions(Crystal::LLVM_VERSION, "11.0.0") >= 0 && !flag?(:darwin) %}
-      LibM.roundeven_f32(self)
-    {% else %}
-      LibM.rint_f32(self)
-    {% end %}
+    # TODO: LLVM 11 introduced llvm.roundeven.* intrinsics which may replace
+    # rint in the future.
+    LibM.rint_f32(self)
   end
 
   # Rounds towards the nearest integer. If both neighboring integers are equidistant,
@@ -255,13 +251,9 @@ struct Float64
   # Rounds towards the nearest integer. If both neighboring integers are equidistant,
   # rounds towards the even neighbor (Banker's rounding).
   def round_even : self
-    # LLVM 11 introduced llvm.roundeven.* intrinsics which may replace rint in
-    # the future.
-    {% if compare_versions(Crystal::LLVM_VERSION, "11.0.0") >= 0 && !flag?(:darwin) %}
-      LibM.roundeven_f64(self)
-    {% else %}
-      LibM.rint_f64(self)
-    {% end %}
+    # TODO: LLVM 11 introduced llvm.roundeven.* intrinsics which may replace
+    # rint in the future.
+    LibM.rint_f64(self)
   end
 
   # Rounds towards the nearest integer. If both neighboring integers are equidistant,

--- a/src/math/libm.cr
+++ b/src/math/libm.cr
@@ -34,10 +34,6 @@ lib LibM
   {% end %}
   fun round_f32 = "llvm.round.f32"(value : Float32) : Float32
   fun round_f64 = "llvm.round.f64"(value : Float64) : Float64
-  {% if compare_versions(Crystal::LLVM_VERSION, "11.0.0") >= 0 %}
-    fun roundeven_f32 = "llvm.roundeven.f32"(value : Float32) : Float32
-    fun roundeven_f64 = "llvm.roundeven.f64"(value : Float64) : Float64
-  {% end %}
   fun rint_f32 = "llvm.rint.f32"(value : Float32) : Float32
   fun rint_f64 = "llvm.rint.f64"(value : Float64) : Float64
   fun sin_f32 = "llvm.sin.f32"(value : Float32) : Float32

